include/xtensor/xnpy.hpp

```diff
@@ -464,8 +467,72 @@ namespace xt
             return header;
         }
 
+
+
     } // namespace detail
 
+    struct npy_file
+    {
+        npy_file() = default;
+
+        npy_file(std::vector<std::size_t>& shape, bool fortran_order,
+                 std::string typestring)
+            : m_shape(shape), m_fortran_order(fortran_order), m_typestring(typestring)
+        {
+            // Allocate memory
+            m_word_size = atoi(&typestring[2]);
```

Comment:
Problem: The "typestring" can have a much more general form where the 3rd byte is not the width.